### PR TITLE
SVG to C converter

### PR DESCRIPTION
**Pull request by cahirwpz**

### CODE REVIEW

File: tools/svg2c/main.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		os.Exit(1)
+	}
+
+	fileInfo, err := os.Stat(inputDir)
+	if err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+
+	if !fileInfo.IsDir() {
```
[the remark this answers is not in the document]
Comment (kjaroma) [reply tools/svg2c/main.go:289]: Well, this was the initial intention but seems to be not the best idea. Will rewrite to accept list of files to merge.

File: tools/svg2c/main.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+}
+
+func handleFile(path string, name string) string {
+	file, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	svgData := SvgData{[]GeometricData{}, Point{0, 0}, cases.Title(language.Und, cases.Compact).String(name)}
```
[the remark this answers is not in the document]
Comment (kjaroma) [reply tools/svg2c/main.go:224]: Yap, not worth to keep extra dependencies

File: tools/svg2c/main.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	for _, point := range gd.Points {
+		withOffset = append(withOffset, Sub(point, offset))
+	}
+	return withOffset
+}
+
+func handleFile(file *os.File, name string) string {
+	svgData := SvgData{[]GeometricData{}, Point{0, 0}, name}
+
+	element, _ := svgparser.Parse(file, false)
```
Comment (spookbench) [inline tools/svg2c/main.go:211]: Why are you discarding an error here?

File: tools/svg2c/main.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+		file, err := os.Open(entry)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer file.Close()
+		fileExt := filepath.Ext(entry)
+		fileName := filepath.Base(entry)
+		if fileExt == ".svg" {
+			if verbose {
+				fmt.Printf("\nConverting file: %s\n", fileName)
```
Comment (spookbench) [inline tools/svg2c/main.go:279]:
```suggestion
				fmt.Println("Converting file:", fileName)
```

File: tools/svg2c/main.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+	svgData.CalcOrigin()
+	svgData.CalcDataWithOffset()
+	data, err := svgData.Export()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return data
+}
+
+const defaultFileName = "data.c"
```
Comment (spookbench) [inline tools/svg2c/main.go:244]: IMO everything above this line should be in a separate file. And if you don't want to do that, then please at least change function's names to lowercase (`CalcDataWithOffset()` or `GetPointsWithOffset()` for example) so they won't be needlessly exported

File: tools/svg2c/main.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+)
+
+type SvgData struct {
+	Data   []GeometricData
+	Origin Point
+	Name   string
+}
+
+type GeometricData struct {
+	Lenght      int
```
Comment (spookbench) [inline tools/svg2c/main.go:47]:
```suggestion
	Length      int
```
:D

**Review by spookbench — COMMENTED** [review]
Mostly stylistic nitpicks  🙂 
I would also consider adding [Go Doc comments](https://tip.golang.org/doc/comment) to the most important functions to explain what's going on. 
One day one of us will have to go back to this code to change it/add a feature and we will be happy to read them 💜

**Pull request merged**